sonic_platform_base/sonic_xcvr/xcvr_api_factory.py

```diff
@@ -34,7 +36,7 @@ def create_xcvr_api(self):
         id = self._get_id()
         # QSFP-DD or OSFP
         if id == 0x18 or id == 0x19:
-            codes = Sff8024
+            codes = {'sff8024':Sff8024, 'cmis_code':CmisCode}
```

Comment:
You can just have 
```suggestion
            codes = CmisCodes
```
once you have it inherit Sff8024.